website/static/js/markdown.js

```diff
@@ -30,12 +31,28 @@ var bootstrapTable = function(md) {
 var oldMarkdownList = function(md) {
     md.block.ruler.after('hr', 'pyMarkdownList', pymarkdownList);
 };
+var mfrURL = window.contextVars.mfrURL;
+var osfURL = window.contextVars.osfURL;
+
+var getMfrUrl = function (guid) {
+    return mfrURL + 'render?url='+ osfURL + guid + '/?action=download%26mode=render';
+};
+
+var mfrId = 0;
 
 // Full markdown renderer for views / wiki pages / pauses between typing
 var markdown = new MarkdownIt('commonmark', {
     highlight: highlighter,
     linkify: true
-})
+    }).use(require('markdown-it-atrules'), {
+        type: 'osf',
+        pattern: /^http(?:s?):\/\/(?:www\.)?[a-zA-Z0-9 .:]{1,}\/render\?url=http(?:s?):\/\/[a-zA-Z0-9 .:]{1,}\/([a-zA-Z0-9]{1,})\/\?action=download|(^[a-zA-Z0-9]{1,}$)/,
+        format: function(assetID) {
+             var id = '__markdown-it-atrules-' + mfrId++;
```

Comment:
@sloria  This works because every time you add a tag it redraws the all other tags, so because mfrId is always incrementing even after you save your changes the ids get redrawn and  you can never repeat the same ID for tags in the same wiki. 

I initially didn't think this would work in all cases, but it seems to work with saved tags and collaborative edits so I'm sticking with this because it seems to guarantee uniqueness.

Reply:
> collaborative edits

Did you test this with multiple users? I would think that this would break with two users, because their counters would both start at 0, so it would be possible to have duplicate IDs.

Reply:
>  because every time you add a tag it redraws the all other tags, 

Ah, I didn't read this closely enough. OK, cool, then yeah, I suppose this would work!

Reply:
^ and that explains why `Date()` didn't work. If everything gets rendered at the same time, then all the timestamps will the same. So `format` *was* getting called multiple times, but it was called multiple times at the same time. Whew, I'm glad there's an explanation.